tasks/offlinePackagingTasks.ts

```diff
@@ -0,0 +1,133 @@
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as debugUtil from '../src/coreclr-debug/util';
+import * as del from 'del';
+import * as fs from 'fs';
+import * as gulp from 'gulp';
+import * as path from 'path';
+import * as unzip from 'unzip2';
+import * as util from '../src/common';
+import spawnNode from '../tasks/spawnNode';
+import { codeExtensionPath, offlineVscodeignorePath, vscodeignorePath, vscePath, packedVsixOutputRoot } from '../tasks/projectPaths';
+import { commandLineOptions } from '../tasks/commandLineArguments';
+import { CsharpLoggerObserver } from '../src/observers/CsharpLoggerObserver';
+import { EventStream } from '../src/EventStream';
+import { getPackageJSON } from '../tasks/packageJson';
+import { Logger } from '../src/logger';
+import { PackageManager } from '../src/packages';
+import { PlatformInformation } from '../src/platform';
+
+gulp.task('vsix:offline:package', () => {
+    del.sync(vscodeignorePath);
```

Comment:
Do we need to delete the vscodeingorePath twice, when we start the task as well as when we exit ?

Reply:
yes. it's a transient build artifact. This is exactly the scenario for which we want the vsce --vscodeignorefile option